devito/arguments.py

```diff
@@ -52,37 +53,38 @@ def as_symbol(self):
         return Symbol(self.name)
 
     @property
-    def ready(self):
-        return self._value is not None
-
-    @property
     def dtype(self):
         return self.provider.dtype
 
     def reset(self):
-        self._value = self._default_value
+        self._value = self.default_value
 
     @abc.abstractproperty
     def verify(self, kwargs):
         return
 
+    def __repr__(self):
+        return self.name
+
 
 class ScalarArgument(Argument):
 
     """ Class representing scalar arguments that a kernel might expect.
         Most commonly used to pass dimension sizes
+        enforce determines whether any reduction will be performed or not.
```

Comment:
use doc-friendly notation:
```
:param enforce: Determine ...
```
Also, add an example? Not straightforward to imagine what's the use case here

Reply:
Q: should we provide a difference type of ``Argument``, rather than adding a parameter such as ``enforce`` ?

Reply:
I don't think your suggestion has anything to do with what is happening here. The argument decides whether the reducer will be called or not. It is still a scalar, but when I am auto-deriving, I want to use a reducer to get the value that everyone agrees on, while if I am using a user-provided value, I want to skip all the reduction operations and just use the provided value directly.